pkg/controller/service/service_controller.go

```diff
@@ -497,7 +497,7 @@ func (s *ServiceController) needsUpdate(oldService *v1.Service, newService *v1.S
 }
 
 func (s *ServiceController) loadBalancerName(service *v1.Service) string {
-	return cloudprovider.GetLoadBalancerName(service)
+	return s.balancer.DefaultLoadBalancerName(service)
```

Comment:
Should be `s.balancer.GetLoadBalancerName(context.TODO(), "", service)`

Reply:
If the `ServiceController` struct defines balancer as `cloudprovider.LoadBalancer`, shouldn't it be `DefaultLoadBalancerName` so that the string gets returned or am I re-implementing GetLoadBalancerName in this file?

Reply:
`s.balancer` implements the LoadBalancer interface so we want to use the new method `GetLoadBalancerName` that we added (which eventually calls DefaultLoadBalancerName)

Reply:
cloudprovider.LoadBalancer here refers to https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/cloud.go#L92

Reply:
I do see the test failing on that though, I may be misreading how ServiceController is used.